src/pkg/services/m365/m365.go

```diff
@@ -29,9 +29,18 @@ type ServiceAccess struct {
 
 // User is the minimal information required to identify and display a user.
 type User struct {
-	PrincipalName string
-	ID            string
-	Name          string
+	PrincipalName                 string
+	ID                            string
+	Name                          string
+	ArchiveFolder                 string
+	Timezone                      string
+	DelegateMeetingMsgDeliveryOpt string
+	DateFormat                    string
+	TimeFormat                    string
+	UserPurpose                   string
+	HasMailBox                    bool
+	HasOnedrive                   bool
+	ErrGettingUserInfo            string
```

Comment:
Why not embed a UserInfo struct here?
```suggestion
    Info UserInfo
```